plugin/defs.go

```diff
@@ -42,6 +44,29 @@ func WithSwarmService(id string) CreateOpt {
 	}
 }
 
+// WithEnv is a CreateOpt that passes the user-provided environment variables
+// to the plugin task
+func WithEnv(env []string) CreateOpt {
+	return func(p *v2.Plugin) {
+		effectiveEnv := make(map[string]string)
+		for _, penv := range p.PluginObj.Config.Env {
+			if penv.Value != nil {
+				effectiveEnv[penv.Name] = *penv.Value
+			}
+		}
+		for _, line := range env {
+			pair := strings.SplitN(line, "=", 2)
+			effectiveEnv[pair[0]] = pair[1]
```

Comment:
This needs a check if `len(pair) > 1`, otherwise it can panic; https://play.golang.org/p/x26T9s-wq1E

```go
package main

import (
	"strings"
)

func main() {
	env := []string{"baz"}
	effectiveEnv := make(map[string]string)
	for _, line := range env {
		pair := strings.SplitN(line, "=", 2)
		effectiveEnv[pair[0]] = pair[1]
	}
}
```

```suggestion
if pair := strings.SplitN(line, "=", 2); len(pair) > 1 {
	effectiveEnv[pair[0]] = pair[1]
}
```